### DISCUSSION TITLE
What LLM framework do you use?

### DISCUSSION THREAD

**Discussion by kleenkanteen** [brianpetro/obsidian-smart-connections#540] [General]
I took a look at `package.json` and found these 3 dependencies:

```
    "smart-chunks": "file:../smart-chunks",
    "smart-collections": "file:../smart-collections",
    "smart-embed": "file:../smart-embed"
```

Can't find those files though. Are you using a LLM framework? If not, why not?

**Comment by brianpetro** [comment]
Hey @kleenkanteen 

All of the files you listed can be found if you click around my GitHub. All are freely available for others to use as minimal dependency modules in their own projects.

> Are you using a LLM framework? If not, why not?

People have also asked me this before, you can find my responses using the search feature to search for "*insert popular VC-backed 'open-source' project name*."

So I added to the README:
*"Minimizing dependencies has been a key principle in the development of Smart Connections."* - https://github.com/brianpetro/obsidian-smart-connections#dependencies

And I made some notes about why I'm keen on minimizing dependencies here https://x.com/wfhbrian/status/1772012079978774548

There are some additional reasons why I believe something like, say, LangChain, would not be best for something like Smart Connections. But I'd rather just go about my way building what I believe is best rather than trying to convince others "why my way's best." Otherwise, I wouldn't be independently (with the help of an incredibly supportive user community) building a freely accessible software that enables individuals with AI, and instead I would be rallying employees & investors to follow my lead as we "build the NBT."

🌴

**Comment by kleenkanteen** [comment]
I see. Thank you for the response!
